core/common/file.hpp

```diff
@@ -33,6 +33,11 @@ namespace fc::common {
     return read(is, span::cast<uint8_t>(values));
   }
 
+  template <typename T>
+  inline bool read1(std::istream &is, T &value) {
```

Comment:
What about more detailed names than read1 & write1 ?

Reply:
"read one", "write one"?

Reply:
why not just overload it?

Reply:
`readByte`?

Reply:
it reads/writes single struct, not "byte".